### PR TITLE
framework/task_manager :  Add README and modify the description in he…

### PR DESCRIPTION
**Pull request by jeongchanKim**
…ader file

1. Add basic description in README.md
2. Modify wrong expression and add description in task_manager.h

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a970745c31a3a4426650dc1321e581b446e06597] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a970745c31a3a4426650dc1321e581b446e06597] - Code Rule Check OK.

File: framework/include/task_manager/task_manager.h [hunk 1]
```diff
@@ -116,8 +116,9 @@ typedef struct task_info_list_s task_info_list_t;
 /**
  * @brief Request to register a task
  * @details @b #include <task_manager/task_manager.h>
- * @param[in] name the name of task which request to register
- * @param[in] permission the permission of task which request to register\n
+ * Only builtin task can be managed by task manager.
+ * @param[in] name the name of task which to be registered
+ * @param[in] permission the permission of task which to be registered\n
```
Comment (sunghan-chang) [inline framework/include/task_manager/task_manager.h]: Above line does not have "\n" at end of line, but here it is. What is right?

File: framework/src/task_manager/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+### Builtin Feature for Registration
+Task manager can manage only builtin applications.  
+Builtin information should follow the [format](https://github.com/Samsung/TizenRT/tree/master/apps/builtin).  
+
+
+Further details are shown below.  
+
+## API List
+[API List](https://github.com/Samsung/TizenRT/tree/master/framework/include/task_manager)
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]:
```
Find APIs' details on header file, [task_manager.h](https://github.com/Samsung/TizenRT/tree/master/framework/include/task_manager/task_manager.h)
```

File: framework/src/task_manager/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Task Manager Sample
+[Task Manager Sample](https://github.com/Samsung/TizenRT/tree/master/apps/examples/task_manager_sample)
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]:
```
## Sample Application
[Task Manager Sample](https://github.com/Samsung/TizenRT/tree/master/apps/examples/task_manager_sample) covers all of above mentioned functionality. Please find sample's details on [README](....../README.md).
```

File: framework/src/task_manager/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+4. provide registered application information APIs  
+    task_manager_getinfo_with_name()  
+    task_manager_getinfo_with_handle()  
+    task_manager_getinfo_with_group()  
+    task_manager_clean_info()  
+    task_manager_clean_infolist()
+
+Please find details in [here](#api-list).
+
+## How to enable
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]:
```
## Prerequisites
```

File: framework/src/task_manager/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# Task Manager
+
+Task manager manages the registered applications.  
+Supported features and APIs are shown below:  
+1. application registration APIs
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]: How about below?
| Features | Functions |
|----------|------------|
| Regitration | task_manger_register() |
|                   | task_manager_unregister() |
|------------|-------------------------------|


Please find https://github.com/Samsung/TizenRT/edit/master/apps/system/utils/README.md

File: framework/include/task_manager/task_manager.h [hunk 6]
```diff
@@ -116,8 +116,9 @@ typedef struct task_info_list_s task_info_list_t;
 /**
  * @brief Request to register a task
  * @details @b #include <task_manager/task_manager.h>
- * @param[in] name the name of task which request to register
- * @param[in] permission the permission of task which request to register\n
+ * Only builtin task can be managed by task manager.
```
Comment (sunghan-chang) [inline framework/include/task_manager/task_manager.h]: Could you leave a link of builtin feature?
```
Find apps/builtin/README.md to know how to use builtin task.
```

**Comment by seinfra** [comment]
Target : [c31555aeb89c2b6b7f81f5e763159e8c191e159f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c31555aeb89c2b6b7f81f5e763159e8c191e159f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [dc483f3e6fc98824e1289cc24ab9ce62c419eea2] -

**Comment by seinfra** [comment]
Target : [dc483f3e6fc98824e1289cc24ab9ce62c419eea2] -

**Comment by seinfra** [comment]
Target : [8c4a08df58d7580577dd10a0ba31c89b5f61e899] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8c4a08df58d7580577dd10a0ba31c89b5f61e899] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cfd12b72d28cf0bae26f95c3bd8a2e2c3c563414] -

**Comment by seinfra** [comment]
Target : [cfd12b72d28cf0bae26f95c3bd8a2e2c3c563414] -

**Comment by seinfra** [comment]
Target : [e2fb86d850d9943b67eab73a8801798f31476fbf] -

**Comment by seinfra** [comment]
Target : [e2fb86d850d9943b67eab73a8801798f31476fbf] -

**Comment by seinfra** [comment]
Target : [0de0d07a137079af6021ae9f129c079aa5f3f8ab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0de0d07a137079af6021ae9f129c079aa5f3f8ab] - Code Rule Check OK.

**Comment by jeongarmy** [comment]
It looks good

**Comment by seinfra** [comment]
Target : [ed05831893730b769f6b1b9bb13851ec9aa2d58c] -

**Comment by seinfra** [comment]
Target : [ed05831893730b769f6b1b9bb13851ec9aa2d58c] -

**Comment by seinfra** [comment]
Target : [5b25be27d9e42903b32000ab1c433bb409c6751f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5b25be27d9e42903b32000ab1c433bb409c6751f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [611858650ed9ff31b04355b2cd96932eee0b438f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [611858650ed9ff31b04355b2cd96932eee0b438f] - Code Rule Check OK.

File: framework/src/task_manager/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+If controller application does not have proper permission for controllee applications, it can't manage the controllee applications.  
+- Permission types
+	- TM_TASK_PERMISSION_ALL : Any Task can request the control to Task Manager  
+	- TM_TASK_PERMISSION_GROUP : Only same group task can request the control to Task Manager  
+		(*group : Tasks set which registered from same parent task)  
+	- TM_TASK_PERMISSION_DEDICATE : Only a task which requested to register that task can request the control to Task Manager
+
+### State
+Each Application can have four states.  
+![State Diagram](media/tm_state_diagram.png)
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]:
```
![State Diagram](https://github.com/Samsung/TizenRT/tree/master/docs/media/tm_state_diagram.png)
```

File: framework/src/task_manager/README.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+All registered applications have their own permission, and the task manager checks the permission for every management request.  
+If controller application does not have proper permission for controllee applications, it can't manage the controllee applications.  
+- Permission types
+	- TM_TASK_PERMISSION_ALL : Any Task can request the control to Task Manager  
+	- TM_TASK_PERMISSION_GROUP : Only same group task can request the control to Task Manager  
+		(*group : Tasks set which registered from same parent task)  
+	- TM_TASK_PERMISSION_DEDICATE : Only a task which requested to register that task can request the control to Task Manager
+
+### State
+Each Application can have four states.
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]: Task Manager manages each application with four states, unregistered, stop, running and paused.
Please find below how to move each state by APIs.

File: framework/src/task_manager/README.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
+# Task Manager
+
+Task manager manages the registered applications.  
+If management between applications is needed, by using the task manager, an application can easily manage other applications.  
+- Example cases  
+ : Bixby can start the Alarm application through the task manager.  
+ : When Wi-Fi disconnection message is broadcasted, applications can receive the broadcast message.
+
+## Features
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]: Details

File: framework/src/task_manager/README.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+Task Manager -> [*] Enable Task Manager
+```
+
+### Builtin Feature for Registration
+Task manager can manage only builtin applications.  
+Builtin information should follow the [format](https://github.com/Samsung/TizenRT/tree/master/apps/builtin).  
+
+
+Further details are shown below.
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]: Let's remove this line

**Comment by seinfra** [comment]
Target : [69f3234d1dbaae9346abb90d12137024ba1157da] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [69f3234d1dbaae9346abb90d12137024ba1157da] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bd8faf797af17b75a953ab48628960001ebeb873] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bd8faf797af17b75a953ab48628960001ebeb873] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [880288370e8fb7d272c3d012393588c590813f2a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [880288370e8fb7d272c3d012393588c590813f2a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7273669bcf4bc0d43563c23d4714c1e4808c465c] -

**Comment by seinfra** [comment]
Target : [7273669bcf4bc0d43563c23d4714c1e4808c465c] -

**Comment by seinfra** [comment]
Target : [5d19f8f52c017aeefcd8e86dd29ecdba9cc7250e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5d19f8f52c017aeefcd8e86dd29ecdba9cc7250e] - Code Rule Check OK.

File: framework/src/task_manager/README.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+Task Manager -> [*] Enable Task Manager
+```
+
+### Builtin Feature for Registration
+Task manager can manage only builtin applications.  
+Builtin information should follow the [format](https://github.com/Samsung/TizenRT/tree/master/apps/builtin).  
+
+
+## Sample Application
+[Task Manager Sample](https://github.com/Samsung/TizenRT/tree/master/apps/examples/task_manager_sample) covers all of above mentioned functionality. Please find sample's details on [README](....../README.md).
```
Comment (sunghan-chang) [inline framework/src/task_manager/README.md]: README link is not working.

**Comment by seinfra** [comment]
Target : [ac952e19c85daffc143d44e0d56fc8cf2d69299e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ac952e19c85daffc143d44e0d56fc8cf2d69299e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3584c45438cfd7bfeb485d2b4019bef40e6f1269] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3584c45438cfd7bfeb485d2b4019bef40e6f1269] - Code Rule Check (C++) OK.

**Pull request merged**